nimare/generate.py

```diff
@@ -105,6 +111,61 @@ def create_coordinate_dataset(
     return ground_truth_foci, dataset
 
 
+def create_neurovault_dataset(
+    collection_ids=NEUROVAULT_IDS,
+    contrasts=CONTRAST_OF_INTEREST,
+    img_dir=None,
+    map_type_conversion=None,
+    **dset_kwargs,
+):
+    """Download images from NeuroVault and use them to create a dataset.
+
+    Parameters
+    ----------
+    collection_ids : :obj:`list` of :obj:`int` or :obj:`dict`, optional
+        A list of collections on neurovault specified by their id.
+        The collection ids can accessed through the neurovault API
+        (i.e., https://neurovault.org/api/collections) or
+        their main website (i.e., https://neurovault.org/collections).
+        For example, in this URL https://neurovault.org/collections/8836/,
+        `8836` is the collection id.
+        collection_ids can also be a dictionary whose keys are the informative
+        study name and the values are collection ids to give the collections
+        human readable names in the dataset.
+    contrasts : :obj:`dict`, optional
+        Dictionary whose keys represent the name of the contrast in
+        the dataset and whose values represent a regular expression that would
+        match the names represented in NeuroVault.
+        For example, under the ``Name`` column in this URL
+        https://neurovault.org/collections/8836/,
+        a valid contrast could be "as-Animal", which will be called "animal" in the created
+        dataset if the contrasts argument is ``{'animal': "as-Animal"}``.
+    img_dir : :obj:`str` or None, optional
+        Base path to save all the downloaded images, by default the images
+        will be saved to a temporary directory with the prefix "neurovault"
+    map_type_conversion : :obj:`dict` or None, optional
+        Dictionary whose keys are what you expect the `map_type` name to
+        be in neurovault and the values are the name of the respective
+        statistic map in a nimare dataset. Default = None.
+    **dset_kwargs : keyword arguments passed to Dataset
+        Keyword arguments to pass in when creating the Dataset object.
+        see :obj:`nimare.dataset.Dataset` for details.
+
+    Returns
+    -------
+    :obj:`nimare.dataset.Dataset`
+        Dataset object containing experiment information from neurovault.
+    """
+    dataset = convert_neurovault_to_dataset(
+        collection_ids, contrasts, img_dir, map_type_conversion, **dset_kwargs
+    )
+    dataset.images = transform_images(
+        dataset.images, target="z", masker=dataset.masker, metadata_df=dataset.metadata
+    )
```

Comment:
Any particular reason only z images are targeted here?

Reply:
z images were targeted because that was the only image type I could reasonably reconstruct from what I downloaded (I only downloaded T images or Z images). Is there another (additional) target that is preferred?

Reply:
That makes sense. I wasn't sure if it would download everything available given the data type dictionary I saw. If the target is generally going to be z images, can you just add a note to the docstring to that effect?

Reply:
Sorry, I don't think I know what you are suggesting to add (to the `convert_neurovault_to_dataset` function docstring?)

The downloader filters based on name of the contrast (e.g., "Working memory load of 2 faces versus 1 face - NT2_Tstat")  and whether that contrast is represented in the `map_type_conversion` dictionary.
If you want to download beta maps and varcopes instead, that is also a valid use case (and closer to what I'm doing for refactoring the meta-analysis for the neuroscout paper.

> If the target is generally going to be z images, can you just add a note to the docstring to that effect?

I'm interpreting this as suggesting `convert_neurovault_to_dataset` will primarily download t or z images, which should not be true since any matching image specified in  `map_type_conversion` should be downloaded, it just so happens with this collection of studies the easiest path forward was to try to download both t and z maps (but each study only has either a z or t map) and convert the studies with t images to z for this analysis.

Reply:
I was just thinking of something like https://github.com/neurostuff/NiMARE/pull/432/files#r588523321. Just some indication about the additional step being taken in this function.